### DISCUSSION TITLE
ZeroTier is my personal VPN - Alan Pope's blog

### DISCUSSION THREAD

**Discussion by giscus** [popey/popey.com-blog#23] [General]
# ZeroTier is my personal VPN - Alan Pope's blog

Back in July, Martin introduced us to ZeroTier on the Linux Matters podcast, episode 8. He detailed why he’s using the tool and how. Worth a listen.
Per their website, ZeroTier “lets you build modern, secure multi-point virtualized networks of almost any type. From robust peer-to-peer networking to multi-cloud mesh infrastructure, we enable global connectivity with the simplicity of a local network.”
Interesting marketing, but do I need this though?

https://popey.com/blog/2023/08/zerotier-is-my-personal-vpn/

**Comment by grifferz** [comment]
And as I commented on Discord after the podcast episode, I think Yggdrasil is better than zerotier. 😀
* It's totally open source and requires no central authority.
* Like zerotier it's a self-organising mesh network so it will communicate point to point over your LAN if both/all devices are local, but as they move away it will retain that connection (using any of your nodes out on the Internet to bridge where necessary). It uses multicast for discovery so you don't need to do anything once you set up a node - it just finds your other nodes.
* It easily allows you to share whole networks, like your LAN, without having to install yggdrasil on every node there.

For some people I think it won't be suitable. That's because:
* It isn't as well documented as zerotier
* Although it's just a single Go binary you can download from GitHub (packaged in Debian/Ubuntu of course), that may be too complicated for people to install on more exotic platforms or it may not support them (not sure; I only do Linux)
* The stable addresses it gives you are IPv6 addresses, so all your apps that you want to use over yggdrasil need to speak IPv6. The communication between nodes itself can use IPv4; you don't need IPv6 to build your network. In fact one of my use cases for yggdrasil is so that my laptop can communicate with my IPv6-only hosts when I'm out and about using strange networks and crappy wifi hotspots that will never have IPv6.

Here's a good write up with some comparisons. It's how I found out about yggdrasil to begin with: https://changelog.complete.org/archives/10478-easily-accessing-all-your-stuff-with-a-zero-trust-mesh-vpn

Oh, I forgot to add: For those who don't need the mesh features and have static networks at either end, and value throughput/performance above all else, pure WireGuard will perform better than zerotier or yggdrasil or anything else like it. So for site to site VPNs probably people should continue doing that. Mesh networks are lots more convenient though, as you found, and at sub-gigabit speeds you probably wouldn't notice.
